### PR TITLE
Add support for building with vcpkg

### PR DESCRIPTION
**Pull request by hallfox**
**Describe your changes**
[vcpkg](https://vcpkg.io/en/) is a useful tool for providing libraries for C++ projects. With a few exceptions for system libraries and bde-tools, this PR adds build presets for CMake that utilize vcpkg as the package provider.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,24 @@ list( APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/etc/cmake" )
...
+  find_package(bal CONFIG PATH_SUFFIXES share/bde REQUIRED)
+  find_package(bdl CONFIG PATH_SUFFIXES share/bde REQUIRED)
+  find_package(nts REQUIRED)
+  find_package(ntc REQUIRED)
+  find_package(benchmark CONFIG REQUIRED)
+  find_package(GTest CONFIG REQUIRED)
+  find_package(ZLIB REQUIRED)
+
+  add_library(benchmark ALIAS benchmark::benchmark)
+  add_library(zlib ALIAS ZLIB::ZLIB)
```
Comment (pniedzielski) [inline CMakeLists.txt:30]: Are the `pkgname::pkgname` target names a VCPKGism?  If so, how come it's just these, and not GTest for instance?  I'd like a short comment right above this line explaining exactly why this is necessary.

File: CMakePresets.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
```
Comment (pniedzielski) [inline CMakePresets.json:1]: Do we need to pull in any of the `bde-tools` toolchain files here, either [by chainloading them](https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/cmake-integration#using-multiple-toolchain-files) or copying some of the configuration into `base`?

File: vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
+  "name": "bmq",
+  "version": "1.0.0",
```
Comment (pniedzielski) [inline vcpkg.json:4]: Should this version match our release version?

File: vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/vcpkg.schema.json",
+  "name": "bmq",
```
Comment (pniedzielski) [inline vcpkg.json:3]: `bmq` -> `blazingmq` ?

**Review by pniedzielski — CHANGES_REQUESTED** [review]
Small changes

File: vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  "version": "1.0.0",
+  "dependencies": [
+    "gtest",
+    "bde",
+    "ntf-core",
+    "benchmark",
+    "zlib",
+    "readline"
```
[the remark this answers is not in the document]
Comment (678098) [reply vcpkg.json:11]: Nevermind, found in comments

File: CMakeLists.txt (hunk 1 above)
Comment (hallfox) [reply CMakeLists.txt:30]: This is how many vcpkg packages choose to export components. It's a common pattern among other uses of CMake. `GTest` as you can tell is not necessary, must have mistakenly added it while I was testing this.

File: CMakePresets.json (hunk 2 above)
Comment (hallfox) [reply CMakePresets.json:1]: No, the toolchains provided by bde-tools haven't been necessary so far. We may consider using chainloading, but what vcpkg's release and override triplet features provide are a fine enough substitute.

File: vcpkg.json (hunk 3 above)
Comment (hallfox) [reply vcpkg.json:4]: Which version? 😛 I'll set it to the bmq SDK version for now, but we should probably have this version match a number that corresponds to some kind of source release.

File: CMakeLists.txt (hunk 1 above)
Comment (pniedzielski) [reply CMakeLists.txt:33]: Resolved in comment.

File: CMakePresets.json (hunk 2 above)
Comment (pniedzielski) [reply CMakePresets.json:1]: Okay, makes sense.

**Review by pniedzielski — APPROVED** [review]

**Review by pniedzielski — APPROVED** [review]

**Pull request merged**